internal/schema/schema.go

```diff
@@ -0,0 +1,50 @@
+package schema
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+// TransformResourceToDataSource is a recursive function that
+// converts an existing Resource schema to a DataSource schema.
+func TransformResourceToDataSource(resourceSchema map[string]*schema.Schema) map[string]*schema.Schema {
+	dataSourceSchema := make(map[string]*schema.Schema, len(resourceSchema))
+
+	for key, definition := range resourceSchema {
+		elementType := definition.Elem
+		isListOrSet := definition.Type == schema.TypeList || definition.Type == schema.TypeSet
+
+		resource, ok := elementType.(*schema.Resource)
+		if ok && isListOrSet {
+			elementType = &schema.Resource{
+				Schema: TransformResourceToDataSource(resource.Schema),
+			}
+		}
+
+		dataSourceSchema[key] = &schema.Schema{
+			Computed:    true,
+			ForceNew:    false,
+			Required:    false,
+			Optional:    false,
+			Description: definition.Description,
+			Type:        definition.Type,
+			Set:         definition.Set,
+			Elem:        elementType,
+		}
+	}
+
+	return dataSourceSchema
+}
+
+// SetExistingAttributesAsOptional updates the schema of existing top level attributes by
+// ensuring they are optional by setting Computed and Required to false and Optional to true.
+func SetExistingAttributesAsOptional(schema map[string]*schema.Schema, keys ...string) {
+	for _, attribute := range keys {
+		if _, ok := schema[attribute]; !ok {
+			break
```

Comment:
Shouldn't this be `continue`?

Reply:
🤦🏻 yes of course! Thanks for spotting it!

Reply:
Should this have been caught by one of the tests?

Reply:
[`86a1f99` (#465)](https://github.com/auth0/terraform-provider-auth0/pull/465/commits/86a1f998b65c62ad6f4c1cf150fd94ca144464e0) updated 👍🏻 

However this would have also gotten caught on the acceptance tests.